src/main/java/ch/njol/skript/expressions/ExprDamageDelay.java

```diff
@@ -0,0 +1,131 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Damage Delay")
+@Description("The time delay before living entities can take damage again.")
+@Examples({
+	"on damage of player:",
+		"\tplayer is holding a diamond named \"Damage saviour\"",
+		"\tadd 2 seconds to the damage delay"
+})
+@Since("INSERT VERSION")
+public class ExprDamageDelay extends SimplePropertyExpression<LivingEntity, Timespan> {
+
+	static {
+		registerDefault(ExprDamageDelay.class, Timespan.class, "[max:max[imum]] (invulnerability|no damage) (tick[s]|delay)", "livingentities");
```

Comment:
Since we're moving to Timespan instead of ticks as a number, and this is already marked as breaking change, the syntax should probably stop using `ticks` as it's misleading and use `time` instead. We've been doing this recently.
 
```suggestion
		registerDefault(ExprDamageDelay.class, Timespan.class, "[max:max[imum]] (invulnerability|no damage) (time|delay)", "livingentities");
```